### PR TITLE
(9139) Update gateway resource to include GCP support

### PR DESCRIPTION
**Pull request by iggynopolis**
1) remove "zone" terraform variable
2) "vpc_reg" and "peering_ha_subnet" to be used for all gateways (including gcp).

### CODE REVIEW

File: aviatrix/resource_gateway.go [hunk 1]
```diff
@@ -316,12 +310,18 @@ func resourceAviatrixGatewayCreate(d *schema.ResourceData, meta interface{}) err
 		LdapPassword:       d.Get("ldap_password").(string),
 		LdapBaseDn:         d.Get("ldap_base_dn").(string),
 		LdapUserAttr:       d.Get("ldap_username_attribute").(string),
-		PeeringHASubnet:    d.Get("peering_ha_subnet").(string),
```
Comment (arvindsree) [inline aviatrix/resource_gateway.go]: Why did you remove this?

Comment (iggynopolis) [reply aviatrix/resource_gateway.go]: PeeringHASubnet doesn't need to be set for the rest api call to create the primary gw,

It's used in a separate rest api call "create_peering_ha_gateway" instead

File: aviatrix/resource_gateway.go [hunk 2]
```diff
@@ -534,16 +538,31 @@ func resourceAviatrixGatewayRead(d *schema.ResourceData, meta interface{}) error
...
+		if gw.CloudType == 1 {
+			// aws vpc_id returns as <vpc_id>~~<other vpc info>
+			d.Set("vpc_id", strings.Split(gw.VpcID, "~~")[0])
+			d.Set("vpc_reg", gw.VpcRegion)
+		} else if gw.CloudType == 4 {
+			// gcp vpc_id returns as <vpc_id>~-~<other vpc info>
+			d.Set("vpc_id", strings.Split(gw.VpcID, "~-~")[0])
+			d.Set("vpc_reg", gw.GatewayZone)
+		} else if gw.CloudType == 8 {
+			d.Set("vpc_id", gw.VpcID)
```
Comment (arvindsree) [inline aviatrix/resource_gateway.go]: Split(gw.VpcID, "~-~")[0]) should work for gcp as well right?

File: aviatrix/resource_gateway.go [hunk 3]
```diff
@@ -534,16 +538,31 @@ func resourceAviatrixGatewayRead(d *schema.ResourceData, meta interface{}) error
...
+			d.Set("vpc_reg", gw.VpcRegion)
+		}
+
 		d.Set("vpc_net", gw.VpcNet)
 		if gw.EnableNat != "" {
 			d.Set("enable_nat", gw.EnableNat)
 		}
-		if gw.AllocateNewEipRead {
+		if gw.AllocateNewEipRead && gw.CloudType == 1 {
```
Comment (arvindsree) [inline aviatrix/resource_gateway.go]: change to nested if for clarity

File: website/docs/r/aviatrix_gateway.html.markdown [hunk 4]
```diff
@@ -24,6 +24,41 @@ resource "aviatrix_gateway" "test_gateway1" {
...
+# Create Aviatrix GCP gateway with Peering HA enabled
+resource "aviatrix_gateway" "test_gateway_gcp" {
+  cloud_type   = 4
+  account_name = "devops-gcp"
+  gw_name      = "avtxgw-gcp"
+  vpc_id       = "gcp-gw-vpc"
+  vpc_reg      = "us-west1-b"
+  vpc_size     = "f1-micro"
+  vpc_net      = "10.12.0.0/24"
+  peering_ha_subnet = "us-west1-c"
```
Comment (arvindsree) [inline website/docs/r/aviatrix_gateway.html.markdown]: Should be CIDR

File: aviatrix/resource_gateway.go [hunk 5]
```diff
@@ -415,10 +415,14 @@ func resourceAviatrixGatewayCreate(d *schema.ResourceData, meta interface{}) err
...
+			GwName:    d.Get("gw_name").(string),
+			CloudType: d.Get("cloud_type").(int),
+		}
+		if peeringHaGateway.CloudType == 1 || peeringHaGateway.CloudType == 8 {
+			peeringHaGateway.PeeringHASubnet = d.Get("peering_ha_subnet").(string)
+		} else if peeringHaGateway.CloudType == 4 {
+			peeringHaGateway.NewZone = d.Get("peering_ha_subnet").(string)
 		}
 		log.Printf("[INFO] Enable peering HA: %#v", peeringHaGateway)
 		err := client.EnablePeeringHaGateway(peeringHaGateway)
```
Comment (arvindsree) [inline aviatrix/resource_gateway.go]: Failure here would cause the state to be empty. You need to set the id after the create step

File: aviatrix/resource_gateway.go (hunk 2 above)
Comment (iggynopolis) [reply aviatrix/resource_gateway.go]: it will still include the first "~" in the vpc_id and cause a diff when applying. Split(gw.VpcID, "~")[0]) works though, would it be better to change to that? I figured that ~-~ would be more exact of a cutoff point, if someone's vpc_id included "~"

File: website/docs/r/aviatrix_gateway.html.markdown (hunk 4 above)
Comment (iggynopolis) [reply website/docs/r/aviatrix_gateway.html.markdown]: For peering_ha_subnet? The rest api asks for a valid GCP zone, so inputting a cidr causes an error.

File: aviatrix/resource_gateway.go (hunk 2 above)
Comment (arvindsree) [reply aviatrix/resource_gateway.go]: ok
